junit5/src/main/java/io/specto/hoverfly/junit5/HoverflyExtensionUtils.java

```diff
@@ -100,13 +101,26 @@ private static void fillHoverflyConfig(io.specto.hoverfly.junit.core.HoverflyCon
         }
     }
 
-    private static SimulationPreprocessorProvider getSimulationPreprocessorProvider(HoverflyConfig configParams) {
+    private static SimulationPreprocessor getSimulationPreprocessor(HoverflyConfig configParams) {
         Class<? extends SimulationPreprocessor> simulationPreprocessorCls = configParams.simulationPreprocessor();
 
         if (UnsetSimulationPreprocessor.class.isAssignableFrom(simulationPreprocessorCls)) {
             return null;
         }
 
-        return SimulationPreprocessorProvider.forClass(simulationPreprocessorCls);
+        try {
+            return ReflectionUtils.newInstance(simulationPreprocessorCls);
+        }
+        catch (Exception ex) {
+            if (ex instanceof NoSuchMethodException) {
+                String message = String.format("Failed to find a no-argument constructor for SimulationPreprocessor [%s]. "
+                                + "Please ensure that a no-argument constructor exists and "
+                                + "that the class is either a top-level class or a static nested class",
+                        simulationPreprocessorCls.getName());
+                throw new HoverflyClientException(message, ex);
```

Comment:
`HoverflyClientException` is reserved for error occurred when calling the Hoverfly Admin API. I don't think we have any existing custom exception that would be suitable here. How about using `IllegalArgumentException`? It could be appropriate for here because user passes in an uninstantiable class.